integration-tests/salesforce/src/main/java/org/apache/camel/quarkus/component/salesforce/SalesforceRoutes.java

```diff
@@ -16,13 +16,66 @@
  */
 package org.apache.camel.quarkus.component.salesforce;
 
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+import javax.ws.rs.Produces;
+
+import io.quarkus.arc.Unremovable;
 import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.salesforce.AuthenticationType;
+import org.apache.camel.component.salesforce.SalesforceComponent;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
 
+@ApplicationScoped
 public class SalesforceRoutes extends RouteBuilder {
 
+    @ConfigProperty(name = "SALESFORCE_USERNAME", defaultValue = "username")
+    String username;
+
+    @ConfigProperty(name = "SALESFORCE_PASSWORD", defaultValue = "password")
+    String password;
+
+    @ConfigProperty(name = "SALESFORCE_CLIENTID", defaultValue = "clientId")
+    String clientId;
+
+    @ConfigProperty(name = "SALESFORCE_CLIENTSECRET", defaultValue = "clientSecret")
+    String clientSecret;
+
+    @Produces
+    @ApplicationScoped
+    @Unremovable
+    @Named("salesforce")
```

Comment:
```suggestion
    @Named("salesforce")
```

You probably want `@javax.enterprise.inject.Produces` instead of `@javax.ws.rs.Produces`. The fact that it works swhows, that both is redundant :). `@ApplicationScoped` is inherited from the class and `@Unremovable` is implicit for `@Named`

Reply:
@ppalaga good catch

Reply:
All the fame belongs to @jamesnetherton from whom I learned all of this recently :)